### PR TITLE
[10.0][ADD] Cronjob to auto update Gravatar

### PR DESCRIPTION
**Pull request by hmrodrigues**
It adds a new field to the user to enable/disable gravatar autoupdate.
This feature can be disabled if the ir.config_parameter gravatar.autoupdate is set to False

Portuguese translations were also updated on base_user_gravatar

Signed-off-by: Hugo Rodrigues <hugorodrigues@openmailbox.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ addon | version | summary
 [base_search_fuzzy](base_search_fuzzy/) | 10.0.1.0.0 | Fuzzy search with the PostgreSQL trigram extension
```
Comment (elicoidal) [inline README.md]: No need to update this file: it is auto updated every night

File: base_user_gravatar/models/res_users.py [hunk 2]
```diff
@@ -39,3 +58,18 @@ def get_gravatar_image(self):
                     )
                 ))
         return True
+
+    @api.model
+    def _update_gravatars(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
+        if auto_update and auto_update.value in ('True', 'true', '1'):
+            _logger.info('Starting gravatar update')
```
Comment (elicoidal) [inline base_user_gravatar/models/res_users.py]: Gravatar

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details. other than LGTM

File: base_user_gravatar/models/res_users.py (hunk 2 above)
Comment (hmrodrigues) [reply base_user_gravatar/models/res_users.py]: Do I a create a second commit or squash into just one?

File: base_user_gravatar/i18n/base_user_gravatar.pot [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Translation of Odoo Server.
```
Comment (Garamotte) [inline base_user_gravatar/i18n/base_user_gravatar.pot]: Remove the pot file.

File: base_user_gravatar/models/res_users.py [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
+        )
+
+    gravatar_autoupdate_enabled = fields.Boolean(
+        compute='_compute_gravatar_autoupdate_enabled'
+        )
+
+    @api.multi
+    def _compute_gravatar_autoupdate_enabled(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
```
Comment (Garamotte) [inline base_user_gravatar/models/res_users.py]: Use `IrParameter.get_param('gravatar.autoupdate')`

File: base_user_gravatar/models/res_users.py [hunk 5]
```diff
@@ -39,3 +58,18 @@ def get_gravatar_image(self):
                     )
                 ))
         return True
+
+    @api.model
+    def _update_gravatars(self):
+        IrParameter = self.env['ir.config_parameter']
+        auto_update = IrParameter.search([('key', '=', 'gravatar.autoupdate')], limit=1)
```
Comment (Garamotte) [inline base_user_gravatar/models/res_users.py]: Use `IrParameter.get_param('gravatar.autoupdate')`

**Review by Garamotte — CHANGES_REQUESTED** [review]

File: base_user_gravatar/i18n/base_user_gravatar.pot (hunk 3 above)
Comment (hmrodrigues) [reply base_user_gravatar/i18n/base_user_gravatar.pot]: It's supposed to leave the old pot file?

Comment (Garamotte) [reply base_user_gravatar/i18n/base_user_gravatar.pot]: No, all pot files are removed from modules in OCA repositories.

**Comment by hmrodrigues** [comment]
I just commit the changes. I don't know if I need to squash into a single commit or if it's ok to have two commits. If I need to squash, please tell me so I can do it.

**Comment by elicoidal** [comment]
@hmrodrigues Better squash the commits whenever possible in order to reduce the footprint

Please check travis too

**Comment by lasley** [comment]
Please don't squash commits into already reviewed commits though- reviewers will typically look at the diffs in order to skip having to recover the full code again. 

Once everything is said and done, the merger can squash on merge, or you can squash at the logical change sets - whichever you prefer.

**Comment by hmrodrigues** [comment]
Regarding pylint, I don't know what to the executable files, are they supposed to be executable? And the error regarding the import error, what should I use?
The line size error, I thought that Odoo ignored PEP8 E501

Sorry about the questions, first time contributing to OCA.

**Comment by moylop260** [comment]
```bash
base_user_gravatar/tests/__init__.py:1: [W8201(incoherent-interpreter-exec-perm), ] You have a python file with execution permissions but you don't have a interpreter magic comment. If you really needs a execution permission then add a magic comment ( https://en.wikipedia.org/wiki/Shebang_(Unix) ). If you don't needs a execution permission then remove it with: chmod -x __init__.py
base_user_gravatar/tests/test_res_users.py:1: [W8201(incoherent-interpreter-exec-perm), ] You have a python file with execution permissions but you don't have a interpreter magic comment. If you really needs a execution permission then add a magic comment ( https://en.wikipedia.org/wiki/Shebang_(Unix) ). If you don't needs a execution permission then remove it with: chmod -x test_res_users.py
```

Run the command `chmod -x base_user_gravatar/tests/test_res_users.py base_user_gravatar/tests/__init__.py`

**Comment by hmrodrigues** [comment]
@moylop260 ok so it doesn't need executable permissions, and about the import error? I don't want to mess up with the tests

**Comment by moylop260** [comment]
```patch
- from odoo.addons.base_user_gravatar.models.res_users import ResUsers
+ from ..models.res_users import ResUsers
```

You can find more OCA guidelines [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#python)

**Comment by hmrodrigues** [comment]
@moylop260 oh thanks

**Comment by hmrodrigues** [comment]
Hey all,
How do I fix the codecov tests? I never used codecov so I don't know how it works.

**Comment by lasley** [comment]
Hi @hmrodrigues - CodeCov isn't testing, but is instead giving an output of how much our test coverage is dropping with the merge of a given PR. I use the [browser plugin](https://docs.codecov.io/docs/browser-extension), which makes it easy to identify hot spots. I'll comment on the misses here inline

File: base_user_gravatar/models/res_users.py [hunk 6]
```diff
@@ -1,19 +1,39 @@
...
+        string='Auto update gravatar'
+        )
+
+    gravatar_autoupdate_enabled = fields.Boolean(
+        compute='_compute_gravatar_autoupdate_enabled'
+        )
+
+    @api.multi
+    def _compute_gravatar_autoupdate_enabled(self):
+        IrParameter = self.env['ir.config_parameter']
```
Comment (lasley) [inline base_user_gravatar/models/res_users.py]: This method is untested

File: base_user_gravatar/models/res_users.py [hunk 7]
```diff
@@ -39,3 +59,19 @@ def get_gravatar_image(self):
                     )
                 ))
         return True
+
+    @api.model
+    def _update_gravatars(self):
+        IrParameter = self.env['ir.config_parameter']
```
Comment (lasley) [inline base_user_gravatar/models/res_users.py:49]: This method is untested

**Pull request closed without merging**